File: jelly-vs-tower/python/tests/tower_builder.py
from __future__ import annotations

from jelly_vs_tower import ColorType, Tower


class TowerBuilder:
    def __init__(self) -> None:
        self._id = "tower-1"
        self._color = ColorType.Blue
        self._level = 1

    def with_id(self, id: str) -> TowerBuilder:
        self._id = id
        return self

    def with_color(self, color: ColorType) -> TowerBuilder:
        self._color = color
        return self

    def with_level(self, level: int) -> TowerBuilder:
        self._level = level
        return self

    def build(self) -> Tower:
        return Tower(self._id, self._color, self._level)

File: jelly-vs-tower/typescript/tests/TowerBuilder.ts
import { ColorType } from '../src/ColorType.js';
import { Tower } from '../src/Tower.js';

export class TowerBuilder {
  private _id = 'tower-1';
  private _color: ColorType = ColorType.Blue;
  private _level = 1;

  withId(id: string): this { this._id = id; return this; }
  withColor(color: ColorType): this { this._color = color; return this; }
  withLevel(level: number): this { this._level = level; return this; }

  build(): Tower {
    return new Tower(this._id, this._color, this._level);
  }
}
